### PR TITLE
Clarify definition of off-path

### PR DESCRIPTION
**Pull request by martinthomson**
We weren't being consistent.  This should help.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -4258,8 +4258,9 @@ Path Maximum Transmission Unit Discovery (PMTUD; {{!RFC1191}}, {{!RFC8201}})
 relies on reception of ICMP messages (e.g., IPv6 Packet Too Big messages) that
 indicate when an IP packet is dropped because it is larger than the local router
 MTU. DPLPMTUD can also optionally use these messages.  This use of ICMP messages
-is potentially vulnerable to off-path attacks that successfully guess the
-addresses used on the path and reduce the PMTU to a bandwidth-inefficient value.
+is potentially vulnerable to attacks by elements that cannot observe packets,
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md:4261]:
```suggestion
is potentially vulnerable to attacks by elements that cannot observe packets
```

File: draft-ietf-quic-tls.md [hunk 2]
```diff
@@ -1386,8 +1386,8 @@ incoming 1-RTT protected packets before the TLS handshake is complete.
 
 Retry packets (see the Retry Packet section of {{QUIC-TRANSPORT}}) carry a
 Retry Integrity Tag that provides two properties: it allows discarding
-packets that have accidentally been corrupted by the network, and it diminishes
-off-path attackers' ability to send valid Retry packets.
+packets that have accidentally been corrupted by the network; and only an
+entity that receives an Initial packet is able to send a valid Retry packet.
```
Comment (janaiyengar) [inline draft-ietf-quic-tls.md:1390]:
```suggestion
packets that have accidentally been corrupted by the network; only an
entity that receives an Initial packet can send a valid Retry packet.
```

**Review by janaiyengar — APPROVED** [review]
This is much better -- thank you @martinduke for catching this.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -4648,8 +4649,8 @@ packet it receives in the Destination Connection ID field.  The value for Source
 Connection ID MUST be copied from the Destination Connection ID of the received
 packet, which is initially randomly selected by a client.  Echoing both
 connection IDs gives clients some assurance that the server received the packet
-and that the Version Negotiation packet was not generated by an off-path
-attacker.
+and that the Version Negotiation packet was not generated by an entity that
+did not receive the Initial packet.
```
Comment (martinduke) [inline draft-ietf-quic-transport.md:4653]:
```suggestion
did not observe the Initial packet.
```

File: draft-ietf-quic-tls.md [hunk 4]
```diff
@@ -1435,7 +1435,8 @@ Original Destination Connection ID:
 : The Original Destination Connection ID contains the value of the Destination
   Connection ID from the Initial packet that this Retry is in response to. The
   length of this field is given in ODCID Length. The presence of this field
-  mitigates an off-path attacker's ability to inject a Retry packet.
+  ensures that a valid Retry packet can only be sent by an entity that
+  receives the Initial packet.
```
Comment (martinduke) [inline draft-ietf-quic-tls.md:1439]:
```suggestion
  observes the Initial packet.
```

File: draft-ietf-quic-tls.md [hunk 5]
```diff
@@ -1386,8 +1386,8 @@ incoming 1-RTT protected packets before the TLS handshake is complete.
 
 Retry packets (see the Retry Packet section of {{QUIC-TRANSPORT}}) carry a
 Retry Integrity Tag that provides two properties: it allows discarding
-packets that have accidentally been corrupted by the network, and it diminishes
-off-path attackers' ability to send valid Retry packets.
+packets that have accidentally been corrupted by the network; only an
+entity that receives an Initial packet can send a valid Retry packet.
```
Comment (martinduke) [inline draft-ietf-quic-tls.md:1390]:
```suggestion
entity that observes an Initial packet can send a valid Retry packet.
```

File: draft-ietf-quic-transport.md [hunk 6]
```diff
@@ -4727,7 +4728,7 @@ In order to prevent tampering by version-unaware middleboxes, Initial packets
 are protected with connection- and version-specific keys (Initial keys) as
 described in {{QUIC-TLS}}.  This protection does not provide confidentiality or
 integrity against on-path attackers, but provides some level of protection
-against off-path attackers.
+against attackers that cannot also receive packets.
```
Comment (martinduke) [inline draft-ietf-quic-transport.md:4731]:
```suggestion
against attackers that cannot also observes packets.
```

File: draft-ietf-quic-transport.md [hunk 7]
```diff
@@ -4727,7 +4728,7 @@ In order to prevent tampering by version-unaware middleboxes, Initial packets
 are protected with connection- and version-specific keys (Initial keys) as
 described in {{QUIC-TLS}}.  This protection does not provide confidentiality or
 integrity against on-path attackers, but provides some level of protection
```
Comment (martinduke) [inline draft-ietf-quic-transport.md:4730]:
```suggestion
integrity against attackers that can observe the packets, but provides some level of protection
```

**Review by martinduke — APPROVED** [review]
I slightly prefer the word "observe" to "receive" because of the connotation, but either way I'm satisfied. Thanks!

File: draft-ietf-quic-transport.md [hunk 8]
```diff
@@ -4726,8 +4727,8 @@ Packet Payload:
 In order to prevent tampering by version-unaware middleboxes, Initial packets
 are protected with connection- and version-specific keys (Initial keys) as
 described in {{QUIC-TLS}}.  This protection does not provide confidentiality or
-integrity against on-path attackers, but provides some level of protection
-against off-path attackers.
+integrity against attackers that can observe packets, but provides some level of
+protection against attackers that cannot also observes packets.
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:4731]:
```suggestion
protection against attackers that cannot observe packets.
```

File: draft-ietf-quic-transport.md [hunk 9]
```diff
@@ -6706,21 +6709,22 @@ An off-path attacker can:
...
+and inject those copies into the network, potentially with spoofed source and
+destination addresses.
 
 For the purposes of this discussion, it is assumed that an off-path attacker
-has the ability to observe, modify, and re-inject a packet into the network
-that will reach the destination endpoint prior to the arrival of the original
-packet observed by the attacker. In other words, an attacker has the ability to
-consistently "win" a race with the legitimate packets between the endpoints,
-potentially causing the original packet to be ignored by the recipient.
+has the ability to observe and re-inject a modified copy of a packet into the
```
Comment (MikeBishop) [inline draft-ietf-quic-transport.md:6722]:
```suggestion
has the ability to inject a modified copy of a packet into the
```
Given that we've just talked about the observe-modify-reinject process in the previous sentence, I think we're safe to shorthand this to injecting a modified packet.

**Review by MikeBishop — APPROVED** [review]
This is an improvement.  Thanks.

**Review by martinduke — APPROVED** [review]

File: draft-ietf-quic-transport.md [hunk 10]
```diff
@@ -4258,8 +4258,9 @@ Path Maximum Transmission Unit Discovery (PMTUD; {{!RFC1191}}, {{!RFC8201}})
 relies on reception of ICMP messages (e.g., IPv6 Packet Too Big messages) that
 indicate when an IP packet is dropped because it is larger than the local router
 MTU. DPLPMTUD can also optionally use these messages.  This use of ICMP messages
-is potentially vulnerable to off-path attacks that successfully guess the
-addresses used on the path and reduce the PMTU to a bandwidth-inefficient value.
+is potentially vulnerable to attacks by elements that cannot observe packets
```
Comment (seanturner) [inline draft-ietf-quic-transport.md:4261]: elements or entities or does it matter? "path elements" is used only once elsewhere in the I-D.

File: draft-ietf-quic-transport.md [hunk 11]
```diff
@@ -6706,19 +6709,19 @@ An off-path attacker can:
 
 An off-path attacker cannot:
 
-- Modify any part of a packet
+- Modify packets sent by endpoints
```
Comment (ekinnear) [inline draft-ietf-quic-transport.md:6712]: Ah, good call

**Review by ekinnear — APPROVED** [review]

**Review by janaiyengar — APPROVED** [review]

**Pull request merged**